src/js/api/core-shim.js

```diff
@@ -293,7 +293,7 @@ function setupError(core, api, error) {
         const contextual = model.get('contextual');
         // Remove (and hide) the player if it failed to set up in contextual mode; otherwise, show the error view
         if (!contextual && !__HEADLESS__) {
-            const errorContainer = ErrorContainer(core, playerError);
+            const errorContainer = ErrorContainer(api, core, playerError);
```

Comment:
Why do we need both the `api` and `core` (controller)? Api is essentially a wrapper for the latter. Events triggered on core/controller are forwarded through the api.

Reply:
I see. It looks like the `core` argument passed into ErrorContainer is treated as a `Model` in which case, I needed the `api` argument to trigger events for the tizen app controls. Happens in view as well
https://github.com/jwplayer/jwplayer/blob/26feb48b314e47172935d5f24f589b918c5b8f3f/src/js/view/view.js#L708

Reply:
Why is `TizenErrorContainer` a child or owned by ErrorContainer? Since it has more responsibility than the standard error ErrorContainer (handles keyboard input) wouldn't it be better it be better for it to use or replace ErrorContainer when needed?